src/synchro.rs

```diff
@@ -610,9 +624,10 @@ macro_rules! resampler_FftFixedin {
 
                 let nbr_chunks_ready =
                     (self.saved_frames as f32 / self.fft_size_in as f32).floor() as usize;
-                let mut wave_out = vec![Vec::new(); self.nbr_channels];
+                wave_out.resize(self.nbr_channels, Vec::new());
```

Comment:
I think this `Vec::new()` allocates even if `wave_out.len() == self.nbr_channels`.

Reply:
This could be avoided by checking if the length of `wave_out` is sufficient before calling Vec::resize. In case `wave_out`'s capacity is too small, allocation has to happen anyway.